src/features/Folding.ts

```diff
@@ -252,15 +256,21 @@ export class FoldingProvider implements vscode.FoldingRangeProvider {
             return 0;
         });
 
+        const settings = this.currentSettings();
         return foldableRegions
             // It's possible to have duplicate or overlapping ranges, that is, regions which have the same starting
             // line number as the previous region. Therefore only emit ranges which have a different starting line
             // than the previous range.
             .filter((item, index, src) => index === 0 || item.startline !== src[index - 1].startline)
             // Convert the internal representation into the VSCode expected type
-            .map((item) => item.toFoldingRange());
+            .map((item) => item.toFoldingRange(settings));
     }
 
+    /**
+     * A helper to return the current extension settings
+     */
+    public currentSettings(): Settings.ISettings { return Settings.load(); }
```

Comment:
Ahhh this all makes sense. Would you able to add a comment with this information? Otherwise I can just imagine someone will try and change it later.

I think the way it currently is makes as much sense as any other way, and is relatively simple as well.

Reply:
Sure... I thought I had, but perhaps I removed it during a rebase. No matter.

Reply:
Fixed.

Reply:
Maybe at some point we could implement DI with features where we could inject the languageService and settingsService into the constructor.  Then we could centrally manage keeping the settings in sync with user changes.  And it would make feature unit testing easier.